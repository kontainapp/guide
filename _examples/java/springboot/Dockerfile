FROM openjdk:11-jdk-slim-buster as build
# RUN mkdir -p /tmp
WORKDIR /opt/src/app

# copy jar
COPY target/spring-boot-hello-1.0.jar app.jar

FROM kontainapp/runenv-jdk-shell-11.0.8:latest as release
COPY --from=build /opt/src /opt/src
# COPY --from=build /tmp /tmp

WORKDIR /opt/src/app

EXPOSE 8080
ENV KM_MGTPIPE=/tmp/km.sock
CMD [ "java", "-XX:-UseCompressedOops", "-jar", "/opt/src/app/app.jar" ]