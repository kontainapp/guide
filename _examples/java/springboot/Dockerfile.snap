# FROM openjdk:11-jdk-slim-buster as build
# RUN mkdir -p /tmp
# WORKDIR /opt/src/app

# # copy jar
# COPY target/spring-boot-hello-1.0.jar app.jar

# FROM kontainapp/runenv-jdk-11 as release
# COPY --from=build /opt/src /opt/src
# COPY --from=build /tmp /tmp

# WORKDIR /opt/src/app

# EXPOSE 8080
# ENV KM_MGTPIPE=/tmp/km.sock
# CMD [ "java", "-XX:-UseCompressedOops", "-jar", "app.jar" ]

FROM kg/springboothello
# FROM kontainapp/spring-boot-demo

ARG TARGET_SNAP=kmsnap
COPY ${TARGET_SNAP} /kmsnap
EXPOSE 8080/tcp
ENV KM_MGTPIPE=/tmp/km.snap_sock
CMD ["/kmsnap"]